tests/test_sagemaker/test_sagemaker_endpoint.py

```diff
@@ -0,0 +1,247 @@
+        instance_type
+    )
+    assert_true(expected_message in e.exception.response["Error"]["Message"])
+
+
+@mock_sagemaker
+def test_create_endpoint():
+    sagemaker = boto3.client("sagemaker", region_name=TEST_REGION_NAME)
+
+    endpoint_name = "MyEndpoint"
+    with assert_raises(ClientError) as e:
+        sagemaker.create_endpoint(
+            EndpointName=endpoint_name, EndpointConfigName="NonexistentEndpointConfig"
+        )
+    assert_true(
+        e.exception.response["Error"]["Message"].startswith(
+            "Could not find endpoint configuration"
+        )
+    )
+
+    model_name = "MyModel"
+    _create_model(sagemaker, model_name)
+
+    endpoint_config_name = "MyEndpointConfig"
+    _create_endpoint_config(sagemaker, endpoint_config_name, model_name)
+
+    resp = sagemaker.create_endpoint(
+        EndpointName=endpoint_name,
+        EndpointConfigName=endpoint_config_name,
+        Tags=GENERIC_TAGS_PARAM,
+    )
+    resp["EndpointArn"].should.match(
+        r"^arn:aws:sagemaker:.*:.*:endpoint/{}$".format(endpoint_name)
+    )
+
+    resp = sagemaker.describe_endpoint(EndpointName=endpoint_name)
+    resp["EndpointArn"].should.match(
+        r"^arn:aws:sagemaker:.*:.*:endpoint/{}$".format(endpoint_name)
+    )
+    resp["EndpointName"].should.equal(endpoint_name)
+    resp["EndpointConfigName"].should.equal(endpoint_config_name)
+    resp["EndpointStatus"].should.equal("InService")
+    assert_true(isinstance(resp["CreationTime"], datetime.datetime))
+    assert_true(isinstance(resp["LastModifiedTime"], datetime.datetime))
+    resp["ProductionVariants"][0]["VariantName"].should.equal("MyProductionVariant")
+
+    resp = sagemaker.list_tags(ResourceArn=resp["EndpointArn"])
+    assert_equal(resp["Tags"], GENERIC_TAGS_PARAM)
+
+
+@mock_sagemaker
+def test_delete_endpoint():
+    sagemaker = boto3.client("sagemaker", region_name=TEST_REGION_NAME)
+
+    model_name = "MyModel"
+    _create_model(sagemaker, model_name)
+
+    endpoint_config_name = "MyEndpointConfig"
+    _create_endpoint_config(sagemaker, endpoint_config_name, model_name)
+
+    endpoint_name = "MyEndpoint"
+    _create_endpoint(sagemaker, endpoint_name, endpoint_config_name)
+
+    sagemaker.delete_endpoint(EndpointName=endpoint_name)
+    with assert_raises(ClientError) as e:
+        sagemaker.describe_endpoint(EndpointName=endpoint_name)
+    assert_true(
+        e.exception.response["Error"]["Message"].startswith("Could not find endpoint")
+    )
+
+    with assert_raises(ClientError) as e:
+        sagemaker.delete_endpoint(EndpointName=endpoint_name)
+    assert_true(
+        e.exception.response["Error"]["Message"].startswith("Could not find endpoint")
+    )
+    pass
+
+
+def _create_model(boto_client, model_name):
+    resp = boto_client.create_model(
+        ModelName=model_name,
+        PrimaryContainer={
+            "Image": "382416733822.dkr.ecr.us-east-1.amazonaws.com/factorization-machines:1",
```

Comment:
You can use the mocked Account ID here, instead of your own (from moto.core import ACCOUNT_ID). 
Same ID is used in the other test file, test_sagemaker_training.py.

Reply:
Hey Bert, that's not a mock image URL, but the URL for one of the AWS pre-built SageMaker container images that one might use for training or an endpoint.  They're region-specific, and you usually you get them from the SageMaker library:

  from sagemaker.amazon.amazon_estimator import get_image_uri
  container = get_image_uri(region_name, 'linear-learner')

I wanted to test with something as close to reality as possible, but not add a dependency on the sagemaker library just for this.